### PR TITLE
Added support for .NET 9

### PR DESCRIPTION
**Pull request by MrDWilson**
Added support for running with .NET 9. A pretty simple update, mainly with some versions bumps and documentation updates. 

One breaking change that had to be accounted for to allow building and test success was implementation of the IModel service, which is now required with the new Entity Framework version. Left as an empty mock for now as it doesn't actually need to be used, just needs to be present.

### CODE REVIEW

**Comment by MrDWilson** [comment]
@rgvlee Codacy issues are resolved now. The need for this is I'm releasing a new service currently that I'd like to have use the latest and greatest (and this is the only package blocking a .NET 9 upgrade), so hope the PR is comprehensive.

**Comment by rgvlee** [comment]
Thanks, the .NET 9 upgrade was on my todo list. I've got some minor changes to do for the .NET 8 version which I am doing right now, and then I'll take a look at these.

**Comment by rgvlee** [comment]
Quick glace most of this looks good, I'll take a closer look at the changes RE `IModel`

File: src/EntityFrameworkCore.Testing.Moq/Extensions/DbContextExtensions.cs [hunk 1]
```diff
@@ -139,10 +140,12 @@ public static TDbContext AddExecuteSqlRawResult<TDbContext>(
...
-                .Returns((string providedSql, IEnumerable<object> providedParameters) => rawSqlCommand)
-                .Callback((string providedSql, IEnumerable<object> providedParameters) =>
+                .Setup(m => m.Build(
+                    It.Is<string>(s => s.Contains(sql, StringComparison.OrdinalIgnoreCase)),
+                    It.Is<IEnumerable<object>>(p => ParameterMatchingHelper.DoInvocationParametersMatchSetUpParameters(parameters, p)),
+                    It.IsAny<IModel>()))
+                .Returns((string _, IEnumerable<object> _, IModel _) => rawSqlCommand)
+                .Callback((string providedSql, IEnumerable<object> providedParameters, IModel _) =>
                 {
                     callback?.Invoke(providedSql, providedParameters);
```
Comment (rgvlee) [inline src/EntityFrameworkCore.Testing.Moq/Extensions/DbContextExtensions.cs:150]: Can't recall why the `callback?.Invoke(providedSql, providedParameters);` is used, but a chance we may need to pass the model
I'll catch this on a subsequent branch

Comment (rgvlee) [reply src/EntityFrameworkCore.Testing.Moq/Extensions/DbContextExtensions.cs:150]: Checked this, looks like a feature I added a while ago, no action required

**Pull request merged**